### PR TITLE
Pfs 52 fee reductions backend

### PR DESCRIPTION
**Pull request by Nicholassully**

### CODE REVIEW

File: finance-api/internal/service/fee_reductions.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	ctx := context.Background()
+
+	feeReductionsRawData, err := s.Store.GetFeeReductions(ctx, pgtype.Int4{Int32: int32(id), Valid: true})
+	if err != nil {
+		return nil, err
+	}
+
+	var feeReductions shared.FeeReductions
+
+	for _, i := range feeReductionsRawData {
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply finance-api/internal/service/fee_reductions.go]: done

File: finance-api/cmd/api/fee_reductions.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"net/http"
+	"strconv"
+)
+
+func (s *Server) getFeeReductions(w http.ResponseWriter, r *http.Request) {
+	clientId, _ := strconv.Atoi(r.PathValue("id"))
+	accountInfo, err := s.Service.GetFeeReductions(clientId)
+
+	if errors.Is(err, pgx.ErrNoRows) {
+		http.Error(w, err.Error(), http.StatusNotFound)
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply finance-api/cmd/api/fee_reductions.go]: done

File: finance-api/internal/store/queries/fee_reductions.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+       datereceived,
+       case
+           when (startdate < CURRENT_DATE and enddate > CURRENT_DATE) then 'Active'
+           when CURRENT_DATE > enddate then 'Expired'
+           else 'Cancelled' end as status,
+       notes
+from fee_reduction fr
+    inner join finance_client fc on fc.id = fr.finance_client_id
+where fr.finance_client_id = $1
+order by enddate desc;
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply finance-api/internal/store/queries/fee_reductions.sql]: done

File: finance-api/internal/store/queries/fee_reductions.sql [hunk 4]
```diff
@@ -0,0 +1,15 @@
+-- name: GetFeeReductions :many
+select fr.id,
+       discounttype,
+       startdate,
+       enddate,
+       datereceived,
+       case
+           when (startdate < CURRENT_DATE and enddate > CURRENT_DATE) then 'Active'
+           when CURRENT_DATE > enddate then 'Expired'
+           else 'Cancelled' end as status,
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply finance-api/internal/store/queries/fee_reductions.sql]: yes I have changed that as well

File: finance-api/internal/store/queries/fee_reductions.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+-- name: GetFeeReductions :many
+select fr.id,
+       discounttype,
+       startdate,
+       enddate,
+       datereceived,
+       case
+           when (startdate < CURRENT_DATE and enddate > CURRENT_DATE) then 'Active'
```
[the remark this answers is not in the document]
Comment (Nicholassully) [reply finance-api/internal/store/queries/fee_reductions.sql]: move out of the sql

File: finance-api/internal/service/fee_reductions_test.go [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+			s := &Service{
+				Store: Store,
+			}
+			got, err := s.GetFeeReductions(tt.id)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("GetFeeReductions() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if (err == nil) && len(*got) == 0 {
+				assert.Empty(t, got)
```
Comment (denialanderror) [inline finance-api/internal/service/fee_reductions_test.go:63]: Same issue here as with the other PR

**Review by denialanderror — CHANGES_REQUESTED** [review]

**Review by denialanderror — APPROVED** [review]

File: finance-api/internal/store/queries/fee_reductions.sql [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+select fr.id,
+       type,
+       startdate,
+       enddate,
+       datereceived,
+       notes,
+       deleted
+from fee_reduction fr
+         inner join finance_client fc on fc.id = fr.finance_client_id
+where fr.finance_client_id = $1
```
Comment (denialanderror) [inline finance-api/internal/store/queries/fee_reductions.sql:11]: Is this correct? The id we are sending down from the FE is the Sirius client ID, not the finance_client id.

**Review by denialanderror — CHANGES_REQUESTED** [review]

File: finance-api/internal/store/queries/fee_reductions.sql (hunk 7 above)
Comment (Nicholassully) [reply finance-api/internal/store/queries/fee_reductions.sql]: Good point changed

**Review by denialanderror — DISMISSED** [review]

**Review by denialanderror — APPROVED** [review]

**Pull request merged**